backend/src/Controllers/elections.controllers.ts

```diff
@@ -1,63 +1,101 @@
 import { Election } from '../../../domain_model/Election';
 import { Ballot } from '../../../domain_model/Ballot';
 import { Score } from '../../../domain_model/Score';
-const ElectionsModel = require('../Models/Elections')
-const BallotModel = require('../Models/Ballots')
+const ElectionsDB = require('../Models/Elections')
 import StarResults from '../StarResults.cjs';
 
+const { Pool } = require('pg');
+const pool = new Pool({
+    connectionString: process.env.DATABASE_URL || 'postgresql://postgres:password@localhost:5432/postgres',
+    ssl:  {
+        rejectUnauthorized: false
+      }
+});
+var ElectionsModel = new ElectionsDB(pool, "electionDB");
+ElectionsModel.init();
+
+
+const electionByID = async (req: any, res: any, next: any) => {
+    try {
+        const election = await ElectionsModel.getElectionByID(parseInt(req.params.id))
+        if (!election)
+            return res.status('400').json({
+                error: "Election not found"
+            })
+        console.log(`Getting Election: ${req.params.id}`)
+        console.log(election)
+        req.election = election
+        next()
+    } catch (err) {
+        return res.status('400').json({
+            error: "Could not retrieve election"
+        })
+    }
+}
 
-const getElection = (req:any, res:any, next:any) => {
-    console.log('Made it to controller')
-    res.json(ElectionsModel.getElectionByID(parseInt(req.params.id)))
-
+const getElection = async (req: any, res: any, next: any) => {
```

Comment:
Is this function still being used?

Reply:
Yes, it is run in the election router 
`router.get('/Election/:id',electionController.getElection)`

getElectionByID is called automatically by the router when the id param is used
`router.param('id', electionController.getElectionByID)`

So when you make a get request to /Election/ID getElectionByID() is run and puts the election inside of req. Then any following middleware functions that need election data has access to it. Then getElection() is called which just moves the election data from req to res. The name of getElection should probably change to be clearer.